tests/test_misc.py

```diff
@@ -17,9 +18,12 @@
 )
 
 
-def test_exit_override():
+def test_exit_override(tmpdir):
```

Comment:
[pytest docs](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures) recommend using the newer fixture `tmp_path`, which is like `tmpdir` but uses `pathlib.Path`.

According to the [`py.path` docs](https://py.readthedocs.io/en/latest/path.html#py-path), it in maintenance mode and not recommended for newer projects. So, this should really be changed to `tmp_path` fixture.